WordPress/WordPressShare/UIImageView+Extensions.swift

```diff
@@ -0,0 +1,122 @@
+import Foundation
+
+
+extension UIImageView
+{
+    /// Downloads an image and updates the UIImageView Instance
+    ///
+    /// - Parameters:
+    ///     - url: The URL of the target image
+    ///     - placeholderImage: Image to be displayed, temporarily, while the Download OP is executed
+    ///
+    public func downloadImage(url: NSURL?, placeholderImage: UIImage?) {
```

Comment:
Why is URL an optional? I feel like we keep having the same argument about this :stuck_out_tongue: 
If `url` is `nil`, it's obvious that this method shouldn't do anything, so it should never be called in the first place.

In this case you could argue that it would at least set the placeholder, but I'd rather do that outside, and in fact we're already doing it if there's no site icon (although we might still get an optional if it's an invalid URL):

```swift
        if let siteIconPath = site.icon, siteIconUrl = NSURL(string: siteIconPath) {
            cell.imageView?.downloadBlavatar(siteIconUrl, placeholderImage: placeholderImage)
        } else {
            cell.imageView?.image = placeholderImage
        }
```

I'd turn that code into this:

```swift
        cell.imageView?.image = placeholderImage
        if let siteIconPath = site.icon, let siteIconUrl = NSURL(string: siteIconPath) {
            cell.imageView?.downloadBlavatar(siteIconUrl)
        }
```

And adjust the `download*` methods to take non-optional `NSURL`